### PR TITLE
Changed medusa-client default request payload to null

### PR DESCRIPTION
**Pull request by nielswart**
## What

changed the medusa-client default request payload to `null`

## Why

Get requests have an empty body and this results in 403 errors see #1043

## How

set default parameter value to `null` and changed type to union of `object` and `null`, i.e `object | null`

## Test

I don't know how to test this?

### CODE REVIEW

**Comment by adrien2p** [comment]
We should privilege Record<string, unknown> over object type.

Don’t use object as a type. The object type is currently hard to use ([see this issue](https://github.com/microsoft/TypeScript/issues/21732)).
Consider using Record<string, unknown> instead, as it allows you to more easily inspect and use the keys.

**Comment by olivermrbl** [comment]
> We should privilege Record<string, unknown> over object type.
> 
> Don’t use object as a type. The object type is currently hard to use ([see this issue](https://github.com/microsoft/TypeScript/issues/21732)). Consider using Record<string, unknown> instead, as it allows you to more easily inspect and use the keys.
> 
> at the moment, the object type is widely used in the project. I think it is something to discuss but if anybody has a counter argument, in those cases Record is better

I like this idea. As far as I can see nothing really speaks for using `object` over `Record<K, V>` right? Record is actually quite nice because it allows us to access object properties without TypeScript complaining.

**Review by srindom — DISMISSED** [review]

File: packages/medusa-js/src/request.ts [hunk 1]
```diff
@@ -183,7 +183,7 @@ class Client {
   async request(
     method: RequestMethod,
     path: string,
-    payload: object = {},
+    payload: object | null = null,
```
Comment (srindom) [inline packages/medusa-js/src/request.ts:186]: As per @adrien2p's suggestion lets do this:

```suggestion
    payload: Record<string, unknown> | null = null,
```

**Review by srindom — CHANGES_REQUESTED** [review]

**Comment by srindom** [comment]
Have tested that the fix works so we just need that small fix and then we can merge

**Review by srindom — APPROVED** [review]

**Comment by adrien2p** [comment]
Yes it is in my branch already

**Comment by srindom** [comment]
@adrien2p @nielswart building `@medusajs/medusa-js` still seems to cause trouble with this update - any ideas? 

Maybe: `Record<symbol, unknown>` is more appropriate?

**Comment by nielswart** [comment]
@srindom @adrien2p how can I help to get this PR completed?

**Comment by adrien2p** [comment]
Hey 👋 did you look at the review comment I suggested ?

**Comment by adrien2p** [comment]
Normally you should see just above something with "has pending changes" no?

**Comment by nielswart** [comment]
> Normally you should see just above something with "has pending changes" no?

Sorry, but I don't see anything on the review?

**Comment by olivermrbl** [comment]
@adrien2p The current suggestion of using `Record<string, unknown>` results in the following error at build time:
`Index signature for type 'string' is missing in type`

Do you happen to know any ways of solving for this?

As far as I can see, the issue is rooted in our types having fixed properties, which does not play nice with Typescript's `Record<string, unknown>` since this indicates that we can have any string as keys in our payloads (which is not really the case either, so using records might be misleading???).

**Comment by olivermrbl** [comment]
Where is this suggestion @adrien2p ? I am also having a hard time finding it 😅

**Comment by adrien2p** [comment]
Line 186 actually

**Comment by olivermrbl** [comment]
Can you throw your suggestion as a comment here? I can't see it :'(

File: packages/medusa-js/src/request.ts [hunk 2]
```diff
@@ -183,7 +183,7 @@ class Client {
   async request(
     method: RequestMethod,
     path: string,
-    payload: Record<string, any> = {},
+    payload: Record<string, unknown> | null = null,
```
[the remark this answers is not in the document]
Comment (adrien2p) [reply packages/medusa-js/src/request.ts:186]: @olivermrbl @nielswart Rho my bad, did not submit the review 😂

[the remark this answers is not in the document]
Comment (nielswart) [reply packages/medusa-js/src/request.ts]: Thanks @olivermrbl and @adrien2p !

**Review by olivermrbl — APPROVED** [review]
LGTM!

**Pull request merged**